README.md

```diff
@@ -1,2 +1,578 @@
+            'beta': results.one.beta
+        }
+
+        // step two
+    }
+```
+
+auto-block allows you to declare all of the mappings in the same place you
+declare the dependencies:
+
+```javascript
+    block = {
+        'one': {
+            func: one
+        },
+        'two': {
+            func: two,
+            with: {
+                'alpha': 'one.alpha',
+                'beta': 'one.beta'
+            }
+        }
+    }
+```
+
+Similarly, controllers often need to detect errors or attaching extra data if
+a step fails:
+
+```javascript
+    async.auto({
+        'one': utility.doStuff,
+        'two': function (results, cb) {
+            var options = {
+                'alpha': results.one.alpha,
+                'beta': results.one.beta
+            }
+            utility.findUser(options, function (err, result) {
+                if (err) {
+                    err.status = 404
+                    err.alpha = options.alpha
+                }
+                cb(err, result);
+            })
+        },
+        'two': ['one', function (results, cb) {
+            // step two
+        }]
+    }, function (err, results) {
+        if (err && err.status) {
+            res.status(err.status).send({
+                message: err.message
+            })
+        }
+    })
+```
+
+auto-block lets you declare these error mappings using `errorDefaults` and
+`errorMappings`:
+
+```javascript
+    var controller: {
+        block: {
+            'one': {
+                func: one
+            },
+            'two': {
+                func: two,
+                with: {
+                    'alpha': 'one.alpha',
+                    'beta': 'one.beta'
+                },
+                errorDefaults: {
+                    'status': 404
+                },
+                errorMappings: {
+                    'alpha': 'one.alpha'
+                }
+            }
+        },
+        done: function (err, results) {
+            if (err && err.status) {
+                res.status(err.status).send({
+                    message: err.message
+                })
+            }
+        }
+    }
+```
+
+These, along with other features, allow you to build complex controllers with
+needing to explicitly write function wrappers. auto-block does all of that for
+you.
+
+Without the need for extra wrappers, the temptation to inline business logic is
+removed. You can comfortably move the business logic out of the controller
+without needing to know the details of which controller module you used.
+
+## Controller configuration
+
+### .done
+
+```javascript
+    var controller = {
+        done: function (error, response) {
+            console.log('do stuff');
+        }
+    }
+```
+
+Called after the entire block has been completed. The values of `error` and
+`response` are generated by their respective mapping configurations (see
+below).
+
+The `done` function can also be provided as the second parameter to `autoBlock.run`:
```

Comment:
Right now, one wins. In the future, I'll add an error warning people not to use both.